service/controller/cluster_resource_set.go

```diff
@@ -441,10 +437,14 @@ func newClusterResourceSet(config clusterResourceSetConfig) (*controller.Resourc
 		updateG8sControlPlanesResource,
 		updateMachineDeploymentsResource,
 		updateInfraRefsResource,
-		keepForInfraRefsResource,
+
+		// Following resources manage CR status information.
+		clusterIDResource,
+		clusterStatusResource,
```

Comment:
These are only managing CR status which does not seem to be relevant at all for the other resources. Moving them here unblocks the others.